src/openrct2/entity/Peep.cpp

```diff
@@ -324,6 +324,12 @@ bool Peep::CheckForPath()
 bool Peep::PathIsBlockedByVehicle()
 {
     auto curPos = TileCoordsXYZ(GetLocation());
+    if (FootpathIsBlockedByVehicle(curPos))
+    {
+        // If current position is blocked, try to get out of the way
+        return false;
+    }
```

Comment:
If footpath is blocked by vehicle (condition), then path is not blocked by vehicle (returns false)? Perhaps the naming could do with some improvement.

Reply:
Perhaps we should rename `Peep::PathIsBlockedByVehicle()` to `Peep::ShouldWaitForTrain()`?

Reply:
I agree, and I prefer the second of @Gymnasiast's suggestions since the level crossing feature isn't technically limited to railways (even though it's the only ride currently supporting it).